packages/next-urql/src/useUrqlValue.ts

```diff
@@ -0,0 +1,46 @@
+'use client';
+
+import React from 'react';
+import { useDataHydrationContext } from './DataHydrationContext';
+import { ssrContext } from './Provider';
+
+export const symbolString = 'urql_transport';
+export const urqlTransportSymbol = Symbol.for(symbolString);
+
+export function useUrqlValue(
+  operationKey: number,
+  value?: { data: any; error: any }
+): void {
+  const ssrExchange = React.useContext(ssrContext);
+  const rehydrationContext = useDataHydrationContext();
+
+  if (!ssrExchange) {
+    throw new Error('forgot to pass an "ssrExchange" to the UrqlProvider.');
+  }
+
+  if (typeof window == 'undefined') {
+    if (rehydrationContext && value) {
+      rehydrationContext.operationValuesByKey[operationKey] = value;
+    }
+  } else {
+    const stores = (window[urqlTransportSymbol as any] ||
```

Comment:
On the client we can have multiple stores, one for each Suspense boundary we have traversed. Eventually we'll need to clean up stores, we could however do something along the lines of just consuming all stores into the ssrExchange and removing them from the window object.